provisioning/README.md

```diff
@@ -0,0 +1,109 @@
+# Mbed Crypto provisioning example application
```

Comment:
The name and the short description of the example are misleading. There are three ways you can provision a device onto a trust network:

* Generate a key off-device, import it onto the device, and enter it into the device database.
* Generate a key on-device, export it from the device, and enter it into the device database.
* Already have a key on the device, export it and enter it into the device database.

(Above, “key” can refer to either the private key or its public key in different parts of the same sentence.)

This application covers both on-device generation and an already-existing key, but not import. This is a useful example, but the description does a bad job of setting the context.

The name is misleading because “provisioning” could apply to a key or to a device. Provisioning a key is not covered at all: that would mean importing a key, which the example does not do. Instead the example uses a key that has already been provisioned. The example performs a _part_ of the process of provisioning a device, so it's misleading to call it just “provisioning”. This is an example of simple (challengeless) **on-boarding**.

Reply:
The example could do importing, but our ATECC608A driver doesn't yet implement this so I had to remove that bit.

Reply:
Another name we use for this sort of application has been "Factory Configuration". I rejected that name because there are multiple factories involved and it wasn't clear which factory was being referenced. We could use with more clear and consistent terminology here.